### ISSUE TITLE
resolveConfig returns wrong TypeScript type for theme keys

### ISSUE THREAD

**Issue by OpenSnack** [tailwindlabs/tailwindcss#9929]
**What version of Tailwind CSS are you using?**

3.2.4

**What build tool (or framework if it abstracts the build tool) are you using?**

Vite 3.2.4

**What version of Node.js are you using?**

16.18.1

**What browser are you using?**

N/A

**What operating system are you using?**

Linux

**Reproduction URL**

[Repro on CodeSandbox](https://codesandbox.io/s/wizardly-chandrasekhar-sk6ln7?file=/src/App.vue) (may take a second or click in the editor for the IDE to pick up the type issue)

**Describe your issue**

Using the [recently exposed resolveConfig types](https://github.com/tailwindlabs/tailwindcss/pull/8924), many config keys (such as `theme.colors`) throw TS errors:

![Unresolved config key type](https://user-images.githubusercontent.com/7517162/203868884-e289f0da-8685-4896-865b-8e3a48772ca3.png)

`ResolvableTo<T>` is an internal type that equals `T | ((utils: PluginUtils) => T)` - presumably a type that gets resolved during `resolveConfig`. The expected colour key (gray, in this case) doesn't exist on the function type, causing the issue. I suspect the resolved config should return the expected `RecursiveKeyValuePair` type with `ResolvableTo` removed.

**Comment by RobinMalfait** [comment]
Hey! Thank you for your bug report!
Much appreciated! 🙏

This should be fixed by #9972, and will be available in the next release.

You can already try it by using the insiders build `npm install tailwindcss@insiders`.

**Comment by curtgrimes** [comment]
I experienced this issue because I did not have [the type annotation in my tailwind.config.js](https://tailwindcss.com/docs/configuration#type-script-types):

```js
/** @type {import('tailwindcss').Config} */
module.exports = {
  content: [
    // ...
  ],
  theme: {
    extend: {},
  },
  plugins: [],
}
```
